include/thread_pool/mpmc_bounded_queue.hpp

```diff
@@ -215,13 +277,10 @@ inline bool MPMCBoundedQueue<T>::pop(T& data)
         intptr_t dif = (intptr_t)seq - (intptr_t)(pos + 1);
         if(dif == 0)
         {
-            if(m_dequeue_pos.compare_exchange_weak(
-                   pos, pos + 1, std::memory_order_relaxed))
-            {
+            if(m_dequeue_pos.compare_exchange_weak(pos, pos + 1, std::memory_order_relaxed))
                 break;
-            }
         }
-        else if(dif < 0)
+        else if(dif < 0 && (!is_strong || m_dequeue_pos.load(std::memory_order_relaxed) == m_enqueue_pos.load(std::memory_order_relaxed)))
```

Comment:
As we spoke this effectively introduces spin locking. Your for loop will end up indefinitely looping if one of the producer threads gets suspended during the process of pushing. And this may have an adverse effect by consuming cpu resources and reducing the chances of the producer thread to wake up.

 This type of behavior is not what you want to be hardwired in the design of your low level LF data structure.

Reply:
Fixed in #3.